src/functions/__tests__/typedEnum.test.ts

```diff
@@ -0,0 +1,143 @@
+import { AssertionError } from 'assert';
+import { typedEnum } from '../typedEnum';
+
+const defaultReportingThreshold = 3;
+
+function runTypedEnum(targetVal: any, reportingThreshold: any) {
+  return typedEnum(
+    targetVal,
+    { reportingThreshold },
+    { given: ['$'] },
+    { given: null, original: null, resolved: {} as any },
+  );
+}
+
+describe('typedEnum', () => {
+  describe('parameters validation', () => {
+    test.each([1, { a: 1 }, 'nope', undefined])(
+      'should return undefined when the enum is not an array (%s)',
+      enumContent => {
+        const schema = {
+          type: 'integer',
+          enum: enumContent,
+        };
+
+        expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+      },
+    );
+
+    test.each([0, -1, { a: 1 }, 'nope', undefined])(
+      'should throw when "reportingThreshold" bears an invalid value (%s)',
+      threshold => {
+        const schema = {
+          type: 'integer',
+          enum: [1, 2, 3],
+        };
+
+        expect(() => runTypedEnum(schema, threshold)).toThrowError(AssertionError);
+      },
+    );
+  });
+
+  test('should return undefined when the enum contains no value', () => {
+    const schema = {
+      type: 'integer',
+      enum: [],
+    };
+
+    expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+  });
+
+  describe('basic', () => {
+    test('should return undefined when all enum values respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 456],
+      };
+
+      expect(runTypedEnum(schema, defaultReportingThreshold)).toBeUndefined();
+    });
+
+    test('should identify enum values that do not respect the format', () => {
+      const schema = {
+        type: 'integer',
+        enum: [123, 'a string!', 456, 'and another one!'],
+      };
+
+      expect(runTypedEnum(schema, defaultReportingThreshold)).toEqual([
+        {
+          message: 'Some enum values do not respect the specified type "integer": "a string!", "and another one!".',
+        },
+      ]);
+    });
+  });
+
+  describe('types', () => {
+    /*
+
+    string (this includes dates and files)
+    number
+    integer
+    boolean
+    array
+    object
+
+    */
+    const testCases: Array<[string, unknown[], unknown]> = [
+      ['string', ['Hello', 'world!'], 12],
+      ['number', [-2147483648, 17.13], 'Hello'],
+      ['integer', [-2147483648, 17], 12.3],
+      ['boolean', [true, false], 1],
+      // array ?
+      // object ?
```

Comment:
@philsturgeon I now realize I've actually never met this use case.

It looks like (from my reading of both oas2 and oas3 specs) that an enum could also typed as  an `array` or an `object`.

Do you confirm this point?

Reply:
OAS 2 and 3 don't have any opinions of their own about what is or is not a valid instance for an enum, but, JSON Schema does: https://json-schema.org/draft/2019-09/json-schema-validation.html#enum

It says it can be used for anything, but I have never see anyone try to do`enum: [[0, 1], [2,3]]` and I don't think we want to waste our time figuring out how to make that happen right now. Let's stick to the cases you've mentioned here, and make sure we test no type (no type = any, so no error)

Reply:
>Let's stick to the cases you've mentioned here
@phil Done.

>make sure we test no type (no type = any, so no error)
Covered by an additional test case